internal/ingress/store/store.go

```diff
@@ -315,7 +328,9 @@ func (s Store) ListGlobalKongPlugins() ([]*configurationv1.KongPlugin, error) {
 		labels.NewSelector().Add(*req),
 		func(ob interface{}) {
 			p, ok := ob.(*configurationv1.KongPlugin)
-			if ok && s.isValidIngresClass(&p.ObjectMeta) {
+			// TODO this isn't in spec, but we're just getting rid of global KongPlugins, correct?
+			// should be possible to just delete this function
```

Comment:
What is not in the spec?

Reply:
For background, this PR originally started as an attempt to address the three issues listed in the original description, two of which were related (both stem from the event handler filters) and one of which (TCPIngress stuff) just kinda got tossed in along with that because it happened to be in the TODO Trello lane.

Basically ignore my use of "spec" here: that change is in the broader class revamp spec, but not part of the individual issues this PR targets--it's more a note from me to myself indicating stuff I need to handle later.

That's perhaps a good opportunity to discuss branching/PR strategy for larger revamps like this that aren't likely to be handled in a single PR, since I agree this one is a bit of a mess.

Reply:
This comment is completely opaque to me.

- If it describes how a chunk of code works (or why it's there) it should not be opaque and we should probably correct the misunderstanding by improving the comment. If the answer is a reference to some legacy/out-of-band decisions, that should be clear from the comment.
- If it does not, and it describes some future plan instead, then it should be a description or a comment on some GitHub issue, imo. Plans change, code remains.

Which case applies here? I suppose it's the latter.

Reply:
If it's the former, my suggested improvement is:

```go
// <some behavior> is not specified in <reference to a spec, e.g. an URL or document title>, but it's here because <reason>
// this function is deprecated because <reason>
```

If it's the latter, my suggested improvement is moving this TODO to a GitHub issue.

Reply:
It's basically just a reminder to myself, since I'm looking at the code more than the issues. Have removed it as I can figure out some way to keep those notes locally.